src/main/java/org/jabref/logic/importer/fetcher/ISIDOREFetcher.java

```diff
@@ -0,0 +1,249 @@
+                .withField(StandardField.ABSTRACT, getAbstract(itemElement.getElementsByTagName("abstract")))
+                .withField(StandardField.DOI, getDOI(itemElement.getElementsByTagName("ore").item(0).getChildNodes()))
+                .withField(StandardField.URL, this.URL);
+    }
+
+    private String getDOI(NodeList list) {
+        for (int i = 0; i < list.getLength(); i++) {
+            if (list.item(i).getTextContent().contains("DOI:")) {
+                return list.item(i).getTextContent().replace("DOI: ", "");
+            }
+            if (list.item(i).getTextContent().contains("doi:")) {
+                return list.item(i).getTextContent().replace("info:doi:", "");
+            }
+        }
+        return "";
+    }
+
+    private String getAbstract(NodeList list) {
+        // If there is only one abstract, return the abstract
+        if (list.getLength() == 1) {
+            return list.item(0).getTextContent();
+        }
+
+        // If there are multiple abstracts, return the one with the english tag
+        for (int i = 0; i < list.getLength(); i++) {
+            Element abstractElement = (Element) list.item(i);
+            String langAttribute = abstractElement.getAttribute("xml:lang");
+            if (Objects.equals(langAttribute, "en")) {
+                return abstractElement.getTextContent();
+            }
+        }
+        return "";
+    }
+
+    // Get the type of the document, ISIDORE only seems to have select types, also their types are different to
+    // those used by Jabaref.
+    private EntryType getType(NodeList list) {
+        for (int i = 0; i < list.getLength(); i++) {
+            String type = list.item(i).getTextContent();
+            if (type.contains("article") || type.contains("Article")) {
+                return StandardEntryType.Article;
+            }
+            if (type.contains("thesis") || type.contains("Thesis")) {
+                return StandardEntryType.Thesis;
+            }
+            if (type.contains("book") || type.contains("Book")) {
+                return StandardEntryType.Book;
+            }
+        }
+        return StandardEntryType.Misc;
+    }
+
+    // Gets all the authors, separated with the word "and"
+    // For some reason the author field sometimes has extra numbers and letters.
+    private String getAuthor(Node itemElement) {
+        StringBuilder stringBuilder = new StringBuilder();
+        for (int i = 1; i < itemElement.getChildNodes().getLength(); i += 2) {
+            String next = removeMultipleSpaces(removeNumbers(itemElement.getChildNodes().item(i).getTextContent()));
+            next = next.replace("\n", "");
+            if (next.isBlank()) {
+                continue;
+            }
+            stringBuilder.append(next);
+            stringBuilder.append(" and ");
+        }
+        return removeMultipleSpaces(stringBuilder.substring(0, stringBuilder.length() - 5)).trim();
```

Comment:
`-5` is a magic number. Please add a constant.

Please read at https://java.by-comparison.com/ for more information.